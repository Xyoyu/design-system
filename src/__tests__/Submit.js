import React from 'react'
import renderer from 'react-test-renderer'
import { Submit } from '..'

describe('Submit', () => {
  test('renders', () => {
    const json = renderer.create(<Submit />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('text-transform', 'uppercase')
  })

  test('renders lg', () => {
    const json = renderer.create(<Submit lg />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
