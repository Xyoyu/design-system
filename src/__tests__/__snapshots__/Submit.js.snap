// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit renders 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.125;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  -webkit-transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  box-shadow: 0 2px 4px rgba(0,0,0,0.125);
  border-radius: 9999px;
  border: 0;
  outline: 0;
  margin: 0px;
  padding-left: 18px;
  padding-right: 18px;
  padding-top: 12px;
  padding-bottom: 12px;
  background-color: #e42d42;
  color: #ffffff;
  font-size: 20px;
}

.c1:hover,
.c1:focus {
  box-shadow: 0 2px 6px rgba(228,45,66,0.25);
}

.c1:active {
  box-shadow: 0 2px 8px 2px rgba(228,45,66,0.25);
}

.c0 {
  text-transform: uppercase;
}

<input
  className="c0 c1"
  color="white"
  fontSize={3}
  type="submit"
/>
`;

exports[`Submit renders lg 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.125;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  -webkit-transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  box-shadow: 0 2px 4px rgba(0,0,0,0.125);
  border-radius: 9999px;
  border: 0;
  outline: 0;
  margin: 0px;
  padding-left: 18px;
  padding-right: 18px;
  padding-top: 12px;
  padding-bottom: 12px;
  background-color: #e42d42;
  color: #ffffff;
  font-size: 20px;
}

.c1:hover,
.c1:focus {
  box-shadow: 0 2px 6px rgba(228,45,66,0.25);
}

.c1:active {
  box-shadow: 0 2px 8px 2px rgba(228,45,66,0.25);
}

.c0 {
  text-transform: uppercase;
}

<input
  className="c0 c1"
  color="white"
  fontSize={3}
  type="submit"
/>
`;
